liquibase-core/src/main/java/liquibase/lockservice/PostgresAdvisoryLockService.java

```diff
@@ -0,0 +1,173 @@
+package liquibase.lockservice;
+
+import liquibase.configuration.GlobalConfiguration;
+import liquibase.configuration.LiquibaseConfiguration;
+import liquibase.database.Database;
+import liquibase.database.core.PostgresDatabase;
+import liquibase.exception.DatabaseException;
+import liquibase.exception.LockException;
+import liquibase.executor.Executor;
+import liquibase.executor.ExecutorService;
+import liquibase.servicelocator.PrioritizedService;
+import liquibase.statement.core.RawSqlStatement;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.logging.Logger;
+
+public class PostgresAdvisoryLockService implements LockService {
+
+    private static final Logger LOGGER = Logger.getLogger(PostgresAdvisoryLockService.class.getName());
+
+    private static final int LOCK_NAMESPACE_ID = "liquibase".hashCode();
+    private static final int LOCK_ID = "lockservice".hashCode();
```

Comment:
Shouldn't the lock ID (or namespace ID) be specific to the schema used, or possibly be made configurable?  Wouldn't this clash if there are multiple schemas supporting different application environments/stages in the same DB instance?

Reply:
For anyone interested, here's my take on this:
https://gist.github.com/stanio/b7843cc18a80b42812ae128aba212245
(I find the Liquibase' own Executor/SqlStatement overly complicated for the task at hand.)

Reply:
> _"Beware of bugs in the above code; I have only proved it correct, not tried it." ⁠—Donald Knuth_

Reply:
There's definitely a use case for multiple Liquibase-managed schemas in the same database.

@berardino, you might be able to use the `oid` of each `DATABASECHANGELOG` table for the advisory lock, instead of a constant hash.

Reply:
> For anyone interested, here's my take on this:
> https://gist.github.com/stanio/b7843cc18a80b42812ae128aba212245
> (I find the Liquibase' own Executor/SqlStatement overly complicated for the task at hand.)

I have now set up a simple project: [liquibase-sessionlock](/stanio/liquibase-sessionlock).

Reply:
That looks great, @stanio, I'll definitely be trying that on my projects. Thanks!

Reply:
FWIW, @blagerweij has made readily available [public packages](https://search.maven.org/artifact/com.github.blagerweij/liquibase-sessionlock), and the project could be found at:

* https://github.com/blagerweij/liquibase-sessionlock